lib/ansible/modules/cloud/amazon/ec2_vpc_igw.py

```diff
@@ -241,9 +241,10 @@ def main():
     state = module.params.get('state', 'present')
     tags = module.params.get('tags')
 
-    nonstring_tags = [k for k, v in tags.items() if not isinstance(v, string_types)]
+    nonstring_tags = [k for k, v in tags.items() if not isinstance(v, (string_types, bool))]
```

Comment:
@aioue Thanks for the patch for this. What do you think about doing away with the valid tags check altogether? Some other opinions on this would probably be good too. I haven't noticed this attempted verification in other modules and I'm not sure what purpose it has. Tags should be able to be integers as well.

Reply:
The input values for tags can be integers but they need to be converted to a string before using e.g. `compare_aws_tags` because they're always strings when returned by boto. I definitely think we should be able to do this in a better way (which is likely just forcing conversion to string, rather than declaring non-string things invalid)

Reply:
@willthames approach appears sound. I couldn't quite work out why it was checking for types in the first place. Casting to string seems like a good way to avoid having to list out every single possible object type.